atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -415,6 +417,26 @@ <T, C extends PreCommitCondition, E extends Exception> T runTaskWithConditionRea
     @Timed
     long getUnreadableTimestamp();
 
+    /**
+     * Returns the timestamp that is before timestamps returned by the consumer on {@link TimestampLockAwareTransaction#preCommit(com.palantir.lock.LockDescriptor, int, Consumer)}
+     * for a {@code timestampLockDescriptor}. This is different from the immutable timestamp because it takes into
+     * account only timestamps returned in that method for the given {@code timestampLockDescriptor},
+     * rather than all start timestamps for open transactions.
+     * <p>
+     * If no transactions with a {@code timestampLockDescriptor} lock are open, this method is
+     * equivalent to {@link Transaction#getTimestamp()}.
+     * <p>
+     * Do consider to fetch the lock timestamp outside of transactions that potentially use it - if fetching the
+     * locked timestamp inside a transaction, it's possible for the locked timestamp to be greater than the
+     * transaction's start timestamp, meaning the transaction cannot read all data up to locked timestamp.
+     *
+     * @param timestampLockDescriptor the string representing the timestampLockDescriptor locks
+     * @return the timestamp that is before any timestamp returned by the consumer of {@link TimestampLockAwareTransaction#preCommit(com.palantir.lock.LockDescriptor, int, Consumer)}
+     */
+    @Beta
+    long getLockedTimestamp(String timestampLockDescriptor);
```

Comment:
I don't like using "lockDescriptor" here, because it kinda implies that this will be the literal name of the lock descriptor. But that's not necessarily true.